src/Domain/Creators/BinaryCompactObjectHelpers.hpp

```diff
@@ -0,0 +1,23 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+
+namespace BinaryCompactObject_detail {
```

Comment:
we've refined our conventions on this: `domain::creators::detail` is what we do nowadays